### ISSUE TITLE
OpenAPIRenderer does not handle API endpoints in multiple namespaces

### ISSUE THREAD

**Issue by Lucidiot** [encode/django-rest-framework#6527]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce

I have made a quick test project to diagnose the problem : https://github.com/Lucidiot/testdrf. Cloning the repo and installing DRF should be all you need. The project has two API views which are held in different URL patterns, which are then included into the root urlconf with two different namespaces.

## Expected behavior

`./manage.py generateschema` should give YAML output holding the paths to all the API endpoints in all namespaces.

## Actual behavior

`./manage.py generateschema` gives a YAML output with zero endpoints ; however using `--format corejson`, it gives a valid CoreJSON file with all the API endpoints. It seems [I am not alone](https://stackoverflow.com/questions/54773880/).

I do not know much about DRF's internals but judging from some quick source code reading and REPL experiments, it *seems* the `SchemaGenerator` gives a `coreapi.Document` with no `links` (an empty OrderedDict) but with `data`. The `OpenAPIRenderer` then does not deal well with this case and simply returns an empty `paths` dict in the YAML output.

**Comment by carltongibson** [comment]
Hi @Lucidiot. 

Can you install the branch from #6532 and input there? We're working on the OpenAPI schema generation right now, so feedback will help make that better. 

I'll leave this open for now because your example project should be helpful.

**Comment by Lucidiot** [comment]
Hi again @carltongibson, it seems that while the OpenAPI schema looks valid, loading it using coreapi gives me a `coreapi.Document` with no `links` again. This could just be an issue on coreapi's side, but I have no idea ; this also could be something that was forgotten about the OpenAPI spec.

I first generated a JSON schema for the `testdrf` project:

``` bash
./manage.py generateschema --format openapi-json > schema.json
```

Then, in a Python shell, I followed the [docs](https://www.django-rest-framework.org/topics/api-clients/#loading-and-saving-schemas) to load a schema:

``` python
>>> from coreapi.codecs import CoreJSONCodec
>>> codec = CoreJSONCodec()
>>> with open('schema.json', 'rb') as f:
...     schema = codec.load(f.read())
...
>>> schema
Document(url='', title='', content={'info': {'title': '', 'version': 'TODO'}, 'openapi': '3.0.2', 'paths': {'/api/v1/test': {'get': {'parameters': [], 'responses': {'200': {'content': {'application/json': {}}}}}}, '/api2/v1/test': {'get': {'parameters': [], 'responses': {'200': {'content': {'application/json': {}}}}}}}})
>>> schema.links
OrderedDict()
>>> schema.data
OrderedDict([('info', Object({'title': '', 'version': 'TODO'})), ('openapi', '3.0.2'), ('paths', Object({'/api/v1/test': {'get': {'parameters': [], 'responses': {'200': {'content': {'application/json': {}}}}}}, '/api2/v1/test': {'get': {'parameters': [], 'responses': {'200': {'content': {'application/json': {}}}}}}}))])
```

**Comment by carltongibson** [comment]
HI @Lucidiot — AFAIK the `coreapi` client doesn't load OpenAPI schemas, which is what's created by the `openapi-json` renderer. 

CoreAPI is being dropped. We're moving to OpenAPI, using the `apistar` library. It's all up in the air at the moment. 

From 3.9 or master: use `coreapi` (Core JSON) and the documented stuff will work. From the #6532 branch it's OpenAPI only (Yaml or JSON) but you'll need to use a OpenAPI compatible tool from there.

**Comment by Lucidiot** [comment]
Oh, okay, sorry! I was getting quite confused there.

**Comment by carltongibson** [comment]
Hi @Lucidiot. Yes, it is confusing right now, not your fault at all. If you can test the dev branch though it'll help make the upcoming 3.10 quicker and better. So it's worth your effort 🙂

**Comment by rpkilby** [comment]
Is this something that can be addressed for 3.10, or should this be punted to 3.11?

**Comment by tomchristie** [comment]
Going to close this off since the 3.10 behavior is OpenAPI based, and it's not obvious that the issue as described applies there. Very happy to reopen if it looks like we *do* still have an issue to deal with, once 3.10 is released.
